apicast/src/provider.lua

```diff
@@ -39,20 +39,32 @@ local _M = {
 }
 
 function _M.configure(contents)
-  local config = configuration.parse(contents)
+  local config, err = configuration.parse(contents)
+
+  if err then
+    ngx.log(ngx.WARN, 'not configured: ', err)
+    return nil, err
+  end
 
   _M.contents = configuration.encode(contents)
-  _M.configured = true
-  _M.configuration = config
-  _M.services = config.services or {} -- for compatibility reasons
+
+  if config then
+    _M.configured = true
+    _M.configuration = config
+    _M.services = config.services or {} -- for compatibility reasons
```

Comment:
yeah, I don't really know :) I guess when testing with system or for liveness/readiness probes of openshift.